### PR TITLE
Add lyra support

### PR DESCRIPTION
**Pull request by yairogen**

### CODE REVIEW

File: queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java [hunk 1]
```diff
@@ -128,31 +136,100 @@ private static void init() {
...
+            final ArrayList<String> serverConnectionKeys = Lists.newArrayList(serverConnections.keySet());
+            Collections.sort(serverConnectionKeys);
 
-            Configuration configuration = ConfigurationFactory.getConfiguration();
-            Configuration subsetBase = configuration.subset("service.rabbitmq");
-            Configuration subsetSecurity = subsetBase.subset("security");
+            Config config = new Config()
+                    .withRecoveryPolicy(new RecoveryPolicy()
+                            .withBackoff(Duration.seconds(1), Duration.seconds(30))
+                            .withMaxAttempts(20))
```
Comment (igreenfield) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: why? or move to configuration, or infinite.

File: queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java [hunk 2]
```diff
@@ -128,31 +136,100 @@ private static void init() {
...
+                    .withConnectionRecoveryPolicy(new RecoveryPolicy().withBackoff(Duration.seconds(1), Duration.seconds(30))
+                            .withMaxAttempts(20))
```
Comment (igreenfield) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: why? or move to configuration, or infinite.

File: queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java (hunk 1 above)
Comment (yairogen) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: you're referring to max attempts?

File: queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java (hunk 2 above)
Comment (yairogen) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: you're referring to max attempts?

File: queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java (hunk 1 above)
Comment (igreenfield) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: Yes

Comment (yairogen) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: please review last commit and let me know if it's ok. 1000 as default with interval of 30 seconds is very high, and high enough IMHO (more than 8 hours)!!!

**Pull request merged**